pandas/core/arrays/string_.py

```diff
@@ -283,7 +283,10 @@ def __setitem__(self, key, value):
         super().__setitem__(key, value)
 
     def fillna(self, value=None, method=None, limit=None):
-        # TODO: validate dtype
+        if value is not None and not (
```

Comment:
you should just be able to call `_validate_setitem_value(value)`, maybe we can it it in the baseclass.

Reply:
i would like this since it would allow us to implement a correct ExtensionBlock._can_hold_element, xref #36226

Reply:
> you should just be able to call `_validate_setitem_value(value)`, maybe we can it it in the baseclass.

Gave that a go - is the last commit what you meant? @jreback @jbrockmendel

Reply:
I am -1 adding it in this PR (or at least adding it on the base class). *If* we add it, it needs to be part of the EA interface, documented that way, have a base extension test, have a fall back implementation, .. (and first discuss if we actually want it).

Also, we actually don't need to explicitly call the setitem validation, since `fillna` *is* already raising that exception because it uses the setitem implementation under the hood. 

So we could also simply add the test to ensure `fillna` raises the proper error.

Reply:
> I am -1 adding it in this PR (or at least adding it on the base class). _If_ we add it to the base class, it needs to be part of the EA interface, documented that way, have a base extension test, have a fall back implementation, .. (and first discuss if we actually want it).

Ok! I'll open an issue

> Also, we actually don't need to explicitly call the setitem validation here, since `fillna` _is_ already raising that exception because it uses the setitem implementation under the hood.
> 
> So we could also simply add the test to ensure `fillna` raises the proper error.

There was an existing test and I added more testcases here